packages/apps/app-dashboard/src/utils/user-dashboard/hexToBase58.ts

```diff
@@ -0,0 +1,28 @@
+import bs58 from 'bs58';
```

Comment:
```
import { arrayify } from 'ethers/lib/utils';
import bs58 from 'bs58';

const hex = '0x1234abcd';
const bytes = arrayify(hex);         // Uint8Array
const base58 = bs58.encode(bytes);   // string
```

arrayify() handles strings with leading 0x or without and gives you a uint8Array which you can then just bs58.encode arrayify will also `throw` if given an empty input